torch/fx/experimental/symbolic_shapes.py

```diff
@@ -4796,11 +4837,17 @@ def track_symfloat(source: Source, val: Union[float, SymFloat]) -> None:
                 if is_dim(source):
                     self.dim_constraints.add_equality(source, expr)
 
-                sexpr = ShapeGuardPrinter(
-                    symbol_to_source, source_ref, self.var_to_sources
-                ).doprint(expr)
-                res = f"{source_ref(source)} == {sexpr}"
-                exprs.append(res)
+                symbol = self.source_to_symbol.get(srcname, sympy.Symbol(srcname))
```

Comment:
This gives me the jeebies but I guess it might be OK if sympy truly can take arbitrarily strings as symbol names. You might want to float this to a prep PR before this one in case some other test failures are from this (concretely, you would split out just this logic for python printing).

Reply:
An alternative is to extend the C++/Python printers to have a few more top level functions: one for equality, another for isnan test, alongside the regular one of printing an arbitrary expression.

Reply:
I moved most of the changes in `symbolic_shapes.py` to https://github.com/pytorch/pytorch/pull/140968 and incorporated your suggestions.